digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/dao/ServicePointBusStopDao.scala

```diff
@@ -0,0 +1,262 @@
+
+case class ServicePoint(id: Long, nationalId: Long, stopTypes: Seq[Int],
+                        municipalityCode: Int, lon: Double, lat: Double,
+                        created: Modification, modified: Modification, propertyData: Seq[Property])
+
+case class ServicePointRow(id: Long, externalId: Long, assetTypeId: Long, point: Option[Point],
+                           validFrom: Option[LocalDate], validTo: Option[LocalDate], property: PropertyRow,
+                           created: Modification, modified: Modification, municipalityCode: Int)
+
+class ServicePointBusStopDao {
+  val logger = LoggerFactory.getLogger(getClass)
+  def typeId: Int = MassTransitStopAsset.typeId
+  val idField = "external_id" //???
+
+  private implicit val getLocalDate = new GetResult[Option[LocalDate]] {
+    def apply(r: PositionedResult) : Option[LocalDate] = {
+      r.nextDateOption().map(new LocalDate(_))
+    }
+  }
+
+  private implicit val getServicePointRow = new GetResult[ServicePointRow] {
+    def apply(r: PositionedResult): ServicePointRow = {
+      val id = r.nextLong
+      val externalId = r.nextLong
+      val assetTypeId = r.nextLong
+      val validFrom = r.nextDateOption.map(new LocalDate(_))
+      val validTo = r.nextDateOption.map(new LocalDate(_))
+      val point = r.nextBytesOption.map(bytesToPoint)
+      val municipalityCode = r.nextInt()
+      val propertyId = r.nextLong
+      val propertyPublicId = r.nextString
+      val propertyType = r.nextString
+      val propertyRequired = r.nextBoolean
+      val propertyMaxCharacters = r.nextIntOption()
+      val propertyValue = r.nextLongOption()
+      val propertyDisplayValue = r.nextStringOption()
+      val property = new PropertyRow(
+        propertyId = propertyId,
+        publicId = propertyPublicId,
+        propertyType = propertyType,
+        propertyRequired = propertyRequired,
+        propertyValue = propertyValue.getOrElse(propertyDisplayValue.getOrElse("")).toString,
+        propertyDisplayValue = propertyDisplayValue.orNull,
+        propertyMaxCharacters = propertyMaxCharacters)
+      val created = new Modification(r.nextTimestampOption().map(new DateTime(_)), r.nextStringOption)
+      val modified = new Modification(r.nextTimestampOption().map(new DateTime(_)), r.nextStringOption)
+
+      ServicePointRow(id, externalId, assetTypeId, point, validFrom, validTo, property, created, modified, municipalityCode = municipalityCode)
+    }
+  }
+
+  def assetRowToProperty(assetRows: Iterable[ServicePointRow]): Seq[Property] = {
+    assetRows.groupBy(_.property.propertyId).map { case (key, rows) =>
+      val row = rows.head
+      Property(
+        id = key,
+        publicId = row.property.publicId,
+        propertyType = row.property.propertyType,
+        required = row.property.propertyRequired,
+        values = rows.map(assetRow =>
+          PropertyValue(
+            assetRow.property.propertyValue,
+            propertyDisplayValueFromAssetRow(assetRow))
```

Comment:
Due the suggestion_box issue and extrations, please change it from:
``` scala
 values = rows.map(assetRow =>
          PropertyValue(
            assetRow.property.propertyValue,
            propertyDisplayValueFromAssetRow(assetRow))
```
To
```scala
values = rows.map(assetRow =>
 val finalValue = PropertyValidator.propertyValueValidation(assetRow.property.publicId, assetRow.property.propertyValue )
    Seq(PropertyValue(finalValue, Option(assetRow.property.propertyDisplayValue)))

Reply:
This makes any sense since all of the assets returned will be a new asset with already with the suggestion box parameter?
This fix is not related to the old assets without that field?

Reply:
After talked with @ricardofmsimoes we agree don't implement this part since all the assets used on this will be new assets and with that, they always will have the new field suggestion box.

This suggestion above make sense in case we have old assets without that property.